index.js

```diff
@@ -10,6 +10,7 @@ var cors = require('cors');
 var createSwaggerObject = require('loopback-swagger').generateSwaggerSpec;
 var SWAGGER_UI_ROOT = require('strong-swagger-ui/index').dist;
 var STATIC_ROOT = path.join(__dirname, 'public');
+var swaggerObject;
```

Comment:
This creates a static global singleton shared by all applications. Consider the following example:

```js
var app1 = loopback();
// setup models, etc.
explorer(app1);

var app2 = loopback();
// setup models, etc.
explorer(app2);
```

With the solution proposed in this pull request, `app1` will serve explorer describing `app2`, which is clearly a bug.

The trick is to keep `swaggerObject` variable defined inside `mountSwagger` and listen for `modelRemoted` event to update it whenever a new model is added.


```js
function mountSwagger(loopbackApplication, swaggerApp, opts) {
  var  swaggerObject = createSwaggerObject(loopbackApplication, opts);

  loopbackApplication.on('modelRemoted', function() {
    swaggerObject = createSwaggerObject(loopbackApplication, opts);
  });

  var resourcePath = opts && opts.resourcePath || 'swagger.json';
  // etc.
}
```